lib/business_logic/dog_list_provider.dart

```diff
@@ -3,19 +3,20 @@ import 'package:array_names/data/repositories/repository.dart';
 import 'package:array_names/service_locator.dart';
 import 'package:flutter_riverpod/flutter_riverpod.dart';
 
-final getDogListProvider = StateNotifierProvider(
+final getDogListProvider = StateNotifierProvider<DogListProvider, List<Breed>>(
```

Comment:
JFYI, could be refactored as
`(ref) => DogListProvider( ref.read(repositoryProvider))
`  
Basically the same code
Looks better, but more complicated to understand